### ISSUE TITLE
Can't set attribute gnote.text

### ISSUE THREAD

**Issue by davenicoll** [djsudduth/keep-it-markdown#22]
Steps to reproduce

1. create a keep note with an image
1. run kim.py to export it
1. line 211 throws an exception - gnote.text = blob_file + "\n" + gnote.text

Workaround

I had to change gnote.text to note_text and write that out.

**Comment by djsudduth** [comment]
Thanks!! I will take a look

**Comment by djsudduth** [comment]
I cannot recreate this problem either. Did you have some unusual characters in your note text?  I tried creating notes with images and all converted without issues - including my normal Keep notes I already had with images.  Are you sure you're using version 0.4.0?

**Comment by davenicoll** [comment]
Yup, I pulled the latest. Seems odd you can't reproduce it, now I'm wondering if it's one of the dependencies. I'll check later today.

**Comment by davenicoll** [comment]
Here are my installed packages
```
click==7.1.2
gkeepapi==0.13.4
  - future [required: >=0.16.0, installed: 0.18.2]
  - gpsoauth [required: >=0.4.1, installed: 0.4.3]
    - pycryptodomex [required: >=3.0, installed: 3.10.1]
    - requests [required: Any, installed: 2.25.1]
      - certifi [required: >=2017.4.17, installed: 2020.12.5]
      - chardet [required: >=3.0.2,<5, installed: 4.0.0]
      - idna [required: >=2.5,<3, installed: 2.10]
      - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.4]
  - six [required: >=1.11.0, installed: 1.15.0]
keyring==23.0.0
  - importlib-metadata [required: >=3.6, installed: 3.7.3]
    - zipp [required: >=0.5, installed: 3.4.1]
  - jeepney [required: >=0.4.2, installed: 0.6.0]
  - SecretStorage [required: >=3.2, installed: 3.3.1]
    - cryptography [required: >=2.0, installed: 3.4.6]
      - cffi [required: >=1.12, installed: 1.14.5]
        - pycparser [required: Any, installed: 2.20]
    - jeepney [required: >=0.6, installed: 0.6.0]
pipdeptree==2.0.0
  - pip [required: >=6.0.0, installed: 20.2.3]
pipreqs==0.4.10
  - docopt [required: Any, installed: 0.6.2]
  - yarg [required: Any, installed: 0.1.9]
    - requests [required: Any, installed: 2.25.1]
      - certifi [required: >=2017.4.17, installed: 2020.12.5]
      - chardet [required: >=3.0.2,<5, installed: 4.0.0]
      - idna [required: >=2.5,<3, installed: 2.10]
      - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.4]
setuptools==49.2.1
```

**Comment by davenicoll** [comment]
I think I've found the problem. Compare these two gnote objects -
```
{
    "_dirty": True,
    "parent": <gkeepapi.node.Root object at 0x7f31359658b0>,
    "id": "14b2229972d.a58510426c442465",
    "server_id": "1U5i5Pua0nnEBGDLxUTwuEpOM6acviyKjYgPAUwd9m-1YqY0HtUxZzaID5o_uMi-aOTzK",
    "parent_id": "root",
    "type": <NodeType.Note: "NOTE">,
    "_sort": "279969792",
    "_version": None,
    "_text": "",
    "_children": {
        "14b2229977a.8a16f6d39e72ad7d": <gkeepapi.node.ListItem object at 0x7f3134c40fa0>
    },
    "timestamps": <gkeepapi.node.NodeTimestamps object at 0x7f3134c39730>,
    "settings": <gkeepapi.node.NodeSettings object at 0x7f3134c39820>,
    "annotations": <gkeepapi.node.NodeAnnotations object at 0x7f3134c39880>,
    "_moved": True,
    "_color": <ColorValue.Blue: "BLUE">,
    "_archived": False,
    "_pinned": False,
    "_title": "Personal bests (PBs)",
    "labels": <gkeepapi.node.NodeLabels object at 0x7f3134c398e0>,
    "collaborators": <gkeepapi.node.NodeCollaborators object at 0x7f3134c39940>
}
```
```
{
    "_dirty": True,
    "parent": <gkeepapi.node.Root object at 0x7fded5b59820>,
    "id": "15f42d9e92b.b4d7ad6675ebfd62",
    "server_id": "1BJmM_K4mtF63DLy7iaf9JYOYeXRCn7kLtU8SbQp8rBtUX8MsIS6upQ-O6npqP-Aetp6k",
    "parent_id": "root",
    "type": <NodeType.List: "LIST">,
    "_sort": "1163919360",
    "_version": None,
    "_text": "",
    "_children": {
        "15f42d9e8cf.88472661aae2f6d7": <gkeepapi.node.Blob object at 0x7fded50eafa0>,
        "15f42d9e92a.85c60e886c2d80ea": <gkeepapi.node.Blob object at 0x7fded50eb160>,
        "15f42e0c306.81eac6345dcb3ba3": <gkeepapi.node.ListItem object at 0x7fded50eb340>,
        "15f42e0c307.9c6416d53ccd5d70": <gkeepapi.node.ListItem object at 0x7fded50eb550>,
        "15f42e0c308.b119f6696421fc9f": <gkeepapi.node.ListItem object at 0x7fded50eb6d0>,
        "15f42e0c309.8a153050e717f5f4": <gkeepapi.node.ListItem object at 0x7fded50eb850>,
        "15f42e0c30a.82275429154fa35c": <gkeepapi.node.ListItem object at 0x7fded50eb9d0>,
        "15f42e0c30b.96c195105cf5319e": <gkeepapi.node.ListItem object at 0x7fded50ebb50>,
        "15f42e0c30c.a3a5aa415c8015d3": <gkeepapi.node.ListItem object at 0x7fded50ebcd0>,
        "15f42e0c30d.b512fa4a0ebaf901": <gkeepapi.node.ListItem object at 0x7fded50ebe50>,
        "15f42e0c30e.b545bccf89ba0716": <gkeepapi.node.ListItem object at 0x7fded50ebfd0>,
        "15f42e0c30f.b24d13968964b6f2": <gkeepapi.node.ListItem object at 0x7fded50ec190>,
        "15f42e0c310.8185d4af2e383e88": <gkeepapi.node.ListItem object at 0x7fded50ec2e0>,
        "15f42e0c311.abdef423d25cb23b": <gkeepapi.node.ListItem object at 0x7fded50ec490>,
        "15f42e0c312.987d32f5d3d9ba8c": <gkeepapi.node.ListItem object at 0x7fded50ec610>,
        "15f42e0c313.bde77ad7ad867d36": <gkeepapi.node.ListItem object at 0x7fded50ec760>,
        "15f42e0c314.9808f905afd15bc8": <gkeepapi.node.ListItem object at 0x7fded50ec910>,
        "15f42e0c315.a1112313480de208": <gkeepapi.node.ListItem object at 0x7fded50eca90>
    },
    "timestamps": <gkeepapi.node.NodeTimestamps object at 0x7fded52ec0a0>,
    "settings": <gkeepapi.node.NodeSettings object at 0x7fded52ec250>,
    "annotations": <gkeepapi.node.NodeAnnotations object at 0x7fded52ec2b0>,
    "_moved": True,
    "_color": <ColorValue.White: "DEFAULT">,
    "_archived": False,
    "_pinned": False,
    "_title": "40 things for 40th",
    "labels": <gkeepapi.node.NodeLabels object at 0x7fded52ec310>,
    "collaborators": <gkeepapi.node.NodeCollaborators object at 0x7fded52ec370>
}
```

The first one works, the second one doesn't. Setting `gnote.text` when the type is `<NodeType.List: 'LIST'>` results in a `can't set attribute` error.

**Comment by djsudduth** [comment]
Thanks - that recreated the error - the LIST type is immutable for the gnote.text field thus throwing the error when attempting to prepend the image markdown (I haven't looked at your fix yet)

**Comment by djsudduth** [comment]
Fix now in 0.4.1 development

**Comment by djsudduth** [comment]
@davenicoll - this was fixed already so I closed your merge.

**Comment by djsudduth** [comment]
released in 0.4.2
